pkg/service/endpoints_controller.go

```diff
@@ -398,32 +390,14 @@ func (e *EndpointController) checkLeftoverEndpoints() {
 	}
 }
 
-func findDefaultPort(pod *api.Pod, servicePort int, proto api.Protocol) int {
-	for _, container := range pod.Spec.Containers {
-		for _, port := range container.Ports {
-			if port.Protocol == proto {
-				return port.ContainerPort
-			}
-		}
-	}
-	return servicePort
-}
-
-// findPort locates the container port for the given manifest and portName.
-// If the targetPort is a non-zero number, use that.  If the targetPort is 0 or
-// not specified, use the first defined port with the same protocol.  If no port
-// is defined, use the service's port.  If the targetPort is an empty string use
-// the first defined port with the same protocol.  If no port is defined, use
-// the service's port.  If the targetPort is a non-empty string, look that
+// findPort locates the container port for the given pod and portName.  If the
```

Comment:
NB: we still have a conflict here if the same port name is used on two containers, but thta is not this PR's problem.